io.openems.edge.battery.api/src/io/openems/edge/battery/statemachine/UndefinedHandler.java

```diff
@@ -0,0 +1,35 @@
+package io.openems.edge.battery.statemachine;
+
+import io.openems.edge.battery.statemachine.StateMachine.State;
+import io.openems.edge.common.statemachine.StateHandler;
+
+public class UndefinedHandler extends StateHandler<State, Context> {
+
+	@Override
+	public State runAndGetNextState(Context context) {
+		var battery = context.getParent();
+		switch (battery.getStartStopTarget()) {
+		case UNDEFINED:
+			// Stuck in UNDEFINED State
+			return State.UNDEFINED;
+
+		case START:
+			// force START
+			if (battery.hasFaults()) {
+				// Has Faults -> error handling
+				return State.ERROR;
+			} else {
+				// No Faults -> start
+				return State.GO_RUNNING;
+			}
+
+		case STOP:
+			// force STOP
+			return State.GO_STOPPED;
+		}
+
+		assert false;
+		return State.UNDEFINED; // can never happen
```

Comment:
you can use the new java 17 switch case expression then you dont need the last 2 lines which might be confusing for some